feature-libs/product-configurator/rulebased/components/group-menu/configurator-group-menu.component.ts

```diff
@@ -290,4 +285,56 @@ export class ConfiguratorGroupMenuComponent {
       })
     );
   }
+
+  /**
+   * Prevents page down behaviour when users press 'ArrowUp' or 'ArrowDown' keys
+   * to change the focus to the corresponding group
+   *
+   * @param {KeyboardEvent} event - Keyboard event
+   */
+  switchGroupOnArrowPress(
+    event: KeyboardEvent,
+    groupIndex: number,
+    group?: Configurator.Group
+  ): void {
+    if (event.code === 'ArrowUp' || event.code === 'ArrowDown') {
+      this.configGroupMenuService.switchGroupOnArrowPress(event, groupIndex);
+    } else if (event.code === 'ArrowLeft' || event.code === 'ArrowRight') {
+      if (this.configGroupMenuService.isBackBtnFocused()) {
+        this.navigateUp();
+      } else if (this.hasSubGroups(group)) {
+        this.click(group);
+      }
+    }
+  }
+
+  containsSelectedGroup(
+    currentGroupId: string,
+    group: Configurator.Group
+  ): boolean {
+    let isCurrentGroupFound = false;
+    if (this.hasSubGroups(group)) {
+      group.subGroups.forEach((subGroup) => {
+        if (this.isGroupSelected(subGroup.id, currentGroupId)) {
+          isCurrentGroupFound = true;
+        }
+      });
+    }
+    return isCurrentGroupFound;
+  }
+
+  setTabIndex(currentGroupId: string, group: Configurator.Group): number {
```

Comment:
nitpick: It doesn't set any value, but returns a value. Let's rename it to getTabIndex